### PR TITLE
Use make-inter-router-requests-sync-fn instead of handle-inter-rou…

### PR DESCRIPTION
**Pull request by shamsimam**
…ter-request-fn in token-reindex-handler-fn

### CODE REVIEW

File: waiter/src/waiter/core.clj [hunk 1]
```diff
@@ -1067,12 +1067,13 @@
...
    :token-reindex-handler-fn (pc/fnk [[:curator kv-store]
-                                      [:routines list-tokens-fn synchronize-fn]
-                                      handle-inter-router-request-fn handle-secure-request-fn]
+                                      [:routines list-tokens-fn make-inter-router-requests-sync-fn synchronize-fn]
+                                      handle-secure-request-fn]
                                (fn token-reindex-handler-fn [request]
                                  (handle-secure-request-fn
                                     (fn inner-token-handler-fn [request]
-                                      (token/handle-reindex-tokens-request synchronize-fn handle-inter-router-request-fn kv-store list-tokens-fn request))
+                                      (token/handle-reindex-tokens-request synchronize-fn make-inter-router-requests-sync-fn
```
Comment (dposada) [inline waiter/src/waiter/core.clj]: What's the functional effect of this change?

Comment (shamsimam) [inline waiter/src/waiter/core.clj]: `make-inter-router-requests-sync-fn` is the crrect function to call, it was a bug in the previous implementation.

Comment (dposada) [inline waiter/src/waiter/core.clj]: What was the bug? Do we need a test?

Comment (dposada) [inline waiter/src/waiter/core.clj]: Discussed in person - this re-index is only triggered manually, which is why this bug was not being triggered

**Pull request merged**